network/framework/src/protocols/health_checker/mod.rs

```diff
@@ -339,10 +340,11 @@ impl<NetworkClient: NetworkClientInterface<HealthCheckerMsg> + Unpin> HealthChec
                     );
                     let peer_network_id =
                         PeerNetworkId::new(self.network_context.network_id(), peer_id);
-                    if let Err(err) = self
-                        .network_interface
-                        .disconnect_peer(peer_network_id)
-                        .await
+                    if let Err(err) = timeout(
```

Comment:
If I understand correctly, this is the actual core of the fix, and the other things are more minor cleanups

Reply:
this is the mitigation, the peer actor part makes sure it won't block forever like it is today